pydevices/acq400Devices/acq400_base.py

```diff
@@ -108,10 +106,62 @@ class _ACQ400_ST_BASE(_ACQ400_BASE):
-        {'path':':SEG_EVENT',   'type':'text',   'value': 'STREAM','options':('no_write_shot',)},
-        {'path':':TRIG_TIME',   'type':'numeric',                  'options':('write_shot',)}
-    ]
+        {'path':':SEG_EVENT',   'type':'text',    'value': 'STREAM','options':('no_write_shot',)},
+        {'path':':TRIG_TIME',   'type':'numeric',                  'options':('write_shot',)},
+        {'path':':DEF_DECIMATE','type':'numeric', 'value': 1,      'options':('no_write_shot',)},
+        #Trigger sources
+        {'path':':TRIG_SRC',   'type':'text',      'value': 'NONE', 'options':('write_shot',)},
+        {'path':':TRIG_DX',    'type':'text',      'value': 'dx', 'options':('write_shot',)},
+        #WRTD tree information
+        {'path':':WR',         'type':'numeric',   'value': 0,  'options':('write_shot',)},
+            {'path':':WR:WRTD_TREE',   'type': 'text', 'value':'WRTD','options': ('no_write_shot')},
+            {'path':':WR:WRTD_T0',     'type': 'numeric', 'value': 0, 'options': ('write_shot')},
+            {'path':':WR:TRIG_TAI',    'type': 'numeric', 'value': 0, 'options': ('write_shot'), 'help': 'The record of the shot number of the wrtd this data came from'},
+        ]
 
+    def init(self):
+        import acq400_hapi
+        # Needs the initialization given in the superclass:
+        super(_ACQ400_ST_BASE, self).init()
+
+        uut = acq400_hapi.Acq400(self.node.data(), monitor=True)
+        
+        # Initializing Sources to NONE:
+        # D0 signal:
+        uut.s0.SIG_SRC_TRG_0   = 'NONE'
+        # D1 signal:
+        uut.s0.SIG_SRC_TRG_1   = 'NONE'
+
+        #Trigger sources choices:
+        # d0:
+        srcs_0 = ['EXT', 'HDMI', 'HOSTB', 'GPG0', 'DSP0', 'WRTT0', 'nc', 'NONE']
+        # d1:
+        srcs_1 = ['STRIG', 'HOSTA', 'HDMI_GPIO', 'GPG1', 'DSP1', 'FP_SYNC', 'WRTT1', 'NONE']
+
+        # When WR is not being used to trigger:
+        if self.wr.data() == 0:
+            # No PRE samples --> EVENT0 doesn't need to be set.
+            # TRIGGER setting
+            if str(self.trig_src.data()) in srcs_1:
+                uut.s0.SIG_SRC_TRG_1   = str(self.trig_src.data())
+                #Setting the signal (dX) to use for ACQ2106 stream control
+                uut.s1.TRG       = 'enable'
+                uut.s1.TRG_DX    = 'd1'
+                uut.s1.TRG_SENSE = 'rising'
+
+            elif str(self.trig_src.data()) in srcs_0:
+                uut.s0.SIG_SRC_TRG_0   = str(self.trig_src.data())
+                #Setting the signal (dX) to use for ACQ2106 stream control
+                uut.s1.TRG       = 'enable'
+                uut.s1.TRG_DX    = 'd0'
+                uut.s1.TRG_SENSE = 'rising'
+            #pass
+        
+        #If WR is used to trigger, then:
+        else:
+            uut.s1.TRG       = 'enable'
+            uut.s1.TRG_DX    = str(self.trig_dx.data())
+            uut.s1.TRG_SENSE = 'rising'
```

Comment:
#127 .. #158 sync_role is meant to hide ALL this stuff.
In the #142 if self.wr.data() == 0: section, sync_role does this NOW

For the #161 else: section,
We owe you a WR-specific sync_role
Also, it's going to be VERY simple, hard codes
uut.s0.SIG_SRC_TRG_1 = WRTT0
uut.s0.SIG_SRC_TRG_1 = WRTT1

=> the higher level UUT-side command saves 27 lines of confusing stuff here..